code/modules/organs/internal/internal_organ_processes.dm

```diff
@@ -0,0 +1,212 @@
+#define BRUISED_2_EFFICIENCY 80
+#define BROKEN_2_EFFICIENCY 50
+#define DEAD_2_EFFICIENCY 0
+
+//Has processes for all internal organs, called from /mob/living/carbon/human/Life()
+
+/mob/living/carbon/human/proc/porcess_internal_ograns() //Calls all of the internal organ processes
+	if(should_have_process(OP_EYES))	//Bad way to do this, should be reworked to call procs from a list that's generated by the species.has_process list + extra processes gotten from organs (carrion)
+		eye_process()
+	if(should_have_process(OP_KIDNEYS))
+		kidney_process()
+	if(should_have_process(OP_LIVER))
+		liver_process()
+	if(should_have_process(OP_HEART))
+		heart_process()
+	if(should_have_process(OP_LUNGS))
+		lung_process()
+	if(is_carrion(src))
+		carrion_process()
+
+/mob/living/carbon/human/proc/get_organ_efficiency(process_define)
+	var/list/process_list = internal_organs_by_efficiency[process_define]
+	var/effective_efficiency = 0
+	if(process_list.len)
+		for(var/organ in process_list)
+			var/obj/item/organ/internal/I = organ
+			effective_efficiency += I.get_process_eficiency(process_define)
+		
+	return effective_efficiency
+
+/mob/living/carbon/human/proc/get_specific_organ_efficiency(process_define, parent_organ_tag)
+	var/effective_efficiency = 0
+	var/obj/item/organ/external/parent_organ = get_organ(parent_organ_tag)
+	for(var/organ in parent_organ.internal_organs)
+		var/obj/item/organ/internal/I = organ
+		if(process_define in I.organ_efficiency)
+			effective_efficiency += I.get_process_eficiency(process_define)
```

Comment:
uhm, this is just a comment but I would make the efficiency of the whole system an average of all the organs that fulfill the same function instead of the sum of the efficiency of all the organs. thus, the average will always be between 0 and 100. but again, it's just a comment I don't know the math you plan to apply for this proc.

Reply:
This should be discussed further.
I don't personally think that the average of all organs efficiencies is a good idea, as it would mean if a single eye is damaged your overall eyesight get's worse and in my opinion, it should still be better (2 healthy eyes VS 3 healthy eyes and 1 damaged eye), but 1 to 1 simple addition probably isn't the best solution either.
I was thinking of maybe having organs outside of their original parent organs only contribute 50% efficiency (eyes in groin) or multiple organs of the same efficiency contributing less than the "main" organ (Long lungs with the highest efficiency 133 (133 base) + other lesser lungs with efficiency of 66 (100 base).
I am open to more ideas.